node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go

```diff
@@ -118,8 +118,26 @@ func (r OsDeviceConnectivityHelperScsiGeneric) RescanDevices(lunId int, arrayIde
 	return nil
 }
 
+/*
+There are two style lun number, one's decimal value is <256 and the other
+is full as 16 hex digit. According to T10 SAM, when decimal value is more
+than 256 and it is converted to the full 16 hex digit, it should be
+swapped and converted into hex.
+
+see the udev function format_lun_number
+https://github.com/systemd/systemd/blob/c4ae2704b7e921a0b05486a7b201be6770a04ea7/src/udev/udev-builtin-path_id.c#L58
```

Comment:
i don't like such links. They may change in the future
f you want to leave it better write in more generic way "github systemd project file udev-builtin-path_id.c"

Reply:
It is a blob object, it won't change.